scheduler/scheduler.go

```diff
@@ -38,7 +38,10 @@ func NewSchedulerService(client clientset.Interface, restclientCfg *restclient.C
 func (s *Service) RestartScheduler(cfg *v1beta2config.KubeSchedulerConfiguration) error {
 	s.ShutdownScheduler()
 
+	oldSchedulerCfg := s.currentSchedulerCfg
 	if err := s.StartScheduler(cfg); err != nil {
+		klog.Info("Applying old configuration setting")
+		s.StartScheduler(oldSchedulerCfg)
```

Comment:
I think it is better to check the error from this second `s.StartScheduler` and the error is printed as warn log when `err != nil` 🙏